oresat_configs/scripts/gen_xtce.py

```diff
@@ -142,247 +152,635 @@ def write_xtce(config: OreSatConfig, dir_path: str = ".") -> None:
+                        size_prefix=res_field.size_prefix,
+                        param_ref=para_ref,
+                    )
+
+                _add_parameter(para_set, para_name, para_type_name, res_field.description)
+                _add_parameter_ref(entry_list, para_name)
+            _add_parameter_ref(entry_list, "hmac")
+            _add_parameter_ref(entry_list, "uslp_fecf")
+
+            cont_ref_entry = ET.SubElement(
+                res_entry_list,
+                "ContainerRefEntry",
                 attrib={
-                    "byteOrder": "leastSignificantByteFirst",
-                    "encoding": encoding,
-                    "sizeInBits": str(DT_LEN[obj.data_type]),
+                    "containerRef": container_name,
                 },
             )
-            if obj.factor != 1:
-                def_cal = ET.SubElement(data_enc, "DefaultCalibrator")
-                poly_cal = ET.SubElement(def_cal, "PolynomialCalibrator")
-                ET.SubElement(
-                    poly_cal,
-                    "Term",
-                    attrib={
-                        "exponent": "1",
-                        "coefficient": str(obj.factor),
-                    },
-                )
-        elif obj.data_type == canopen.objectdictionary.VISIBLE_STRING:
-            para_type = ET.SubElement(
-                tm_meta_para,
-                "StringParameterType",
+            inc_cond = ET.SubElement(cont_ref_entry, "IncludeCondition")
+            ET.SubElement(
+                inc_cond,
+                "Comparison",
                 attrib={
-                    "name": name,
+                    "parameterRef": "edl_command_code",
+                    "value": str(cmd.uid),
                 },
             )
-            str_para_type = ET.SubElement(
-                para_type,
-                "StringDataEncoding",
+
+
+def gen_xtce(args: Optional[Namespace] = None) -> None:
+    """Gen_dcf main."""
+    if args is None:
+        args = build_parser(ArgumentParser()).parse_args()
+
+    config = OreSatConfig(args.oresat)
+    write_xtce(config, args.dir_path)
+
+
+def _add_parameter_type(
+    root,
+    name: str,
+    data_type: str,
+    description: str = "",
+    unit: str = "",
+    factor: float = 1,
+    default: Any = None,
+    value_descriptions: dict[str, int] = {},
+    size_prefix: int = 0,
+    param_ref: str = "",
+):
+
+    if data_type == "bool":
+        para_type = ET.SubElement(
+            root,
+            "BooleanParameterType",
+            attrib={
+                "name": name,
+                "zeroStringValue": "0",
+                "oneStringValue": "1",
+            },
+        )
+    elif data_type.startswith("uint") and value_descriptions:  # enums
+        para_type = ET.SubElement(
+            root,
+            "EnumeratedParameterType",
+            attrib={
+                "name": name,
+            },
+        )
```

Comment:
Both `BooleanParameterType` and `EnumeratedParameterType` require `IntegerDataEncoding`, which It looks like is `<IntegerDataEncoding encoding="unsigned" sizeInBits="8" />` for all of them.

Reply:
Fixed